From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mio <mio@mio19.uk>
Date: Tue, 20 Oct 2026 11:10:35 +1100
Subject: [PATCH] devices: import AsteroidOS smartwatch device catalog

This patch imports configurations for the AsteroidOS smartwatch catalog:

- Asus ZenWatch series: asus-anthias, asus-sparrow, asus-swift, and asus-wren.
- LG watch series: lg-bass, lg-dory, lg-lenok, lg-narwhal, and lg-nemo.
- Mobvoi TicWatch series: ticwatch-catfish, ticwatch-harmony, ticwatch-inharmony,
  ticwatch-mooneye, ticwatch-rubyfish, ticwatch-skipjack, and ticwatch-tunny.
- Motorola Moto 360 series: motorola-minnow and motorola-smelt.
- Other watches: casio-koi, oppo-beluga, polar-pike, samsung-rinato,
  samsung-sprat, and sony-tetra.

Signed-off-by: Mio <mio@mio19.uk>
---
 devices/asus-anthias/default.nix       | 57 ++++++++++++++++++++++++++
 devices/asus-sparrow/default.nix       | 56 +++++++++++++++++++++++++
 devices/asus-swift/default.nix         | 56 +++++++++++++++++++++++++
 devices/asus-wren/default.nix          | 57 ++++++++++++++++++++++++++
 devices/casio-koi/default.nix          | 55 +++++++++++++++++++++++++
 devices/lg-bass/default.nix            | 55 +++++++++++++++++++++++++
 devices/lg-dory/default.nix            | 57 ++++++++++++++++++++++++++
 devices/lg-lenok/default.nix           | 55 +++++++++++++++++++++++++
 devices/lg-narwhal/default.nix         | 56 +++++++++++++++++++++++++
 devices/lg-nemo/default.nix            | 56 +++++++++++++++++++++++++
 devices/motorola-minnow/default.nix    | 55 +++++++++++++++++++++++++
 devices/motorola-smelt/default.nix     | 57 ++++++++++++++++++++++++++
 devices/oppo-beluga/default.nix        | 57 ++++++++++++++++++++++++++
 devices/polar-pike/default.nix         | 54 ++++++++++++++++++++++++
 devices/samsung-rinato/default.nix     | 54 ++++++++++++++++++++++++
 devices/samsung-sprat/default.nix      | 57 ++++++++++++++++++++++++++
 devices/sony-tetra/default.nix         | 54 ++++++++++++++++++++++++
 devices/ticwatch-catfish/default.nix   | 56 +++++++++++++++++++++++++
 devices/ticwatch-harmony/default.nix   | 54 ++++++++++++++++++++++++
 devices/ticwatch-inharmony/default.nix | 54 ++++++++++++++++++++++++
 devices/ticwatch-mooneye/default.nix   | 54 ++++++++++++++++++++++++
 devices/ticwatch-rubyfish/default.nix  | 56 +++++++++++++++++++++++++
 devices/ticwatch-skipjack/default.nix  | 56 +++++++++++++++++++++++++
 devices/ticwatch-tunny/default.nix     | 56 +++++++++++++++++++++++++
 24 files changed, 1334 insertions(+)
 create mode 100644 devices/asus-anthias/default.nix
 create mode 100644 devices/asus-sparrow/default.nix
 create mode 100644 devices/asus-swift/default.nix
 create mode 100644 devices/asus-wren/default.nix
 create mode 100644 devices/casio-koi/default.nix
 create mode 100644 devices/lg-bass/default.nix
 create mode 100644 devices/lg-dory/default.nix
 create mode 100644 devices/lg-lenok/default.nix
 create mode 100644 devices/lg-narwhal/default.nix
 create mode 100644 devices/lg-nemo/default.nix
 create mode 100644 devices/motorola-minnow/default.nix
 create mode 100644 devices/motorola-smelt/default.nix
 create mode 100644 devices/oppo-beluga/default.nix
 create mode 100644 devices/polar-pike/default.nix
 create mode 100644 devices/samsung-rinato/default.nix
 create mode 100644 devices/samsung-sprat/default.nix
 create mode 100644 devices/sony-tetra/default.nix
 create mode 100644 devices/ticwatch-catfish/default.nix
 create mode 100644 devices/ticwatch-harmony/default.nix
 create mode 100644 devices/ticwatch-inharmony/default.nix
 create mode 100644 devices/ticwatch-mooneye/default.nix
 create mode 100644 devices/ticwatch-rubyfish/default.nix
 create mode 100644 devices/ticwatch-skipjack/default.nix
 create mode 100644 devices/ticwatch-tunny/default.nix

diff --git a/devices/asus-anthias/default.nix b/devices/asus-anthias/default.nix
new file mode 100644
index 00000000..0d0d7d5e
--- /dev/null
+++ b/devices/asus-anthias/default.nix
@@ -0,0 +1,57 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "asus-anthias";
+  mobile.device.identity = {
+    name = "ZenWatch 1";
+    manufacturer = "ASUS";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon 400 (APQ8026)
+    soc = "qualcomm-apq8026";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 320;
+      shape = "square";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "anthias";
+    bootimg.flash = {
+      offset_base = "0x00000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=anthias"
+    "msm_rtb.filter=0x237"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/asus-sparrow/default.nix b/devices/asus-sparrow/default.nix
new file mode 100644
index 00000000..cad2f302
--- /dev/null
+++ b/devices/asus-sparrow/default.nix
@@ -0,0 +1,56 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "asus-sparrow";
+  mobile.device.identity = {
+    name = "ZenWatch 2";
+    manufacturer = "ASUS";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon 400 (APQ8026)
+    soc = "qualcomm-apq8026";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 320;
+      shape = "square";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "sparrow";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=sparrow"
+    "msm_rtb.filter=0x237"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/asus-swift/default.nix b/devices/asus-swift/default.nix
new file mode 100644
index 00000000..1b8520fd
--- /dev/null
+++ b/devices/asus-swift/default.nix
@@ -0,0 +1,56 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "asus-swift";
+  mobile.device.identity = {
+    name = "ZenWatch 3";
+    manufacturer = "ASUS";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon Wear 2100 (MSM8909W)
+    soc = "qualcomm-msm8909w";
+    ram = 512;
+    screen = {
+      width = 400;
+      height = 400;
+      shape = "round";
+    };
+    watch = {
+      hasCrown = true;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "swift";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=swift"
+    "msm_rtb.filter=0x237"
+  ];
+}
diff --git a/devices/asus-wren/default.nix b/devices/asus-wren/default.nix
new file mode 100644
index 00000000..5e306742
--- /dev/null
+++ b/devices/asus-wren/default.nix
@@ -0,0 +1,57 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "asus-wren";
+  mobile.device.identity = {
+    name = "ZenWatch 2 (42mm)";
+    manufacturer = "ASUS";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon 400 (APQ8026)
+    soc = "qualcomm-apq8026";
+    ram = 512;
+    screen = {
+      width = 280;
+      height = 280;
+      shape = "square";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "wren";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=wren"
+    "msm_rtb.filter=0x237"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/casio-koi/default.nix b/devices/casio-koi/default.nix
new file mode 100644
index 00000000..e5661f5d
--- /dev/null
+++ b/devices/casio-koi/default.nix
@@ -0,0 +1,55 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "casio-koi";
+  mobile.device.identity = {
+    name = "Smart Outdoor Watch (WSD-F10/F20)";
+    manufacturer = "Casio";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Samsung Exynos 3250
+    soc = "samsung-exynos3250";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 300;
+      shape = "chin";
+      chinHeight = 20;
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = false;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "koi";
+    bootimg.flash = {
+      offset_base = "0x10000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttySAC2,115200"
+    "androidboot.hardware=koi"
+  ];
+}
diff --git a/devices/lg-bass/default.nix b/devices/lg-bass/default.nix
new file mode 100644
index 00000000..ee5e7150
--- /dev/null
+++ b/devices/lg-bass/default.nix
@@ -0,0 +1,55 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "lg-bass";
+  mobile.device.identity = {
+    name = "Watch Urbane";
+    manufacturer = "LG";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon 400 (APQ8026)
+    soc = "qualcomm-apq8026";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 320;
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "bass";
+    bootimg.flash = {
+      offset_base = "0x00000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=bass"
+    "msm_rtb.filter=0x237"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/lg-dory/default.nix b/devices/lg-dory/default.nix
new file mode 100644
index 00000000..4627627e
--- /dev/null
+++ b/devices/lg-dory/default.nix
@@ -0,0 +1,57 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "lg-dory";
+  mobile.device.identity = {
+    name = "G Watch";
+    manufacturer = "LG";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon 400 (APQ8026)
+    soc = "qualcomm-apq8026";
+    ram = 512;
+    screen = {
+      width = 280;
+      height = 280;
+      shape = "square";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = false;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "dory";
+    bootimg.flash = {
+      offset_base = "0x00000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=dory"
+    "msm_rtb.filter=0x237"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/lg-lenok/default.nix b/devices/lg-lenok/default.nix
new file mode 100644
index 00000000..7ac3aab1
--- /dev/null
+++ b/devices/lg-lenok/default.nix
@@ -0,0 +1,55 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "lg-lenok";
+  mobile.device.identity = {
+    name = "G Watch R";
+    manufacturer = "LG";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon 400 (APQ8026)
+    soc = "qualcomm-apq8026";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 320;
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "lenok";
+    bootimg.flash = {
+      offset_base = "0x00000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=lenok"
+    "msm_rtb.filter=0x237"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/lg-narwhal/default.nix b/devices/lg-narwhal/default.nix
new file mode 100644
index 00000000..5142fbdd
--- /dev/null
+++ b/devices/lg-narwhal/default.nix
@@ -0,0 +1,56 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "lg-narwhal";
+  mobile.device.identity = {
+    name = "Watch W7";
+    manufacturer = "LG";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon Wear 2100 (MSM8909W)
+    soc = "qualcomm-msm8909w";
+    ram = 768;
+    screen = {
+      width = 360;
+      height = 360;
+      shape = "round";
+    };
+    watch = {
+      hasCrown = true;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = false;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "narwhal";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=narwhal"
+    "msm_rtb.filter=0x237"
+  ];
+}
diff --git a/devices/lg-nemo/default.nix b/devices/lg-nemo/default.nix
new file mode 100644
index 00000000..083b4a92
--- /dev/null
+++ b/devices/lg-nemo/default.nix
@@ -0,0 +1,56 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "lg-nemo";
+  mobile.device.identity = {
+    name = "Watch Urbane 2nd Edition";
+    manufacturer = "LG";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon 400 (MSM8926)
+    soc = "qualcomm-msm8926";
+    ram = 768;
+    screen = {
+      width = 480;
+      height = 480;
+      shape = "round";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "nemo";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=nemo"
+    "msm_rtb.filter=0x237"
+  ];
+}
diff --git a/devices/motorola-minnow/default.nix b/devices/motorola-minnow/default.nix
new file mode 100644
index 00000000..41e8e99e
--- /dev/null
+++ b/devices/motorola-minnow/default.nix
@@ -0,0 +1,55 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "motorola-minnow";
+  mobile.device.identity = {
+    name = "Moto 360 (2014)";
+    manufacturer = "Motorola";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # TI OMAP 3630
+    soc = "ti-omap3630";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 290;
+      shape = "chin";
+      chinHeight = 30;
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = false;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "minnow";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyO2,115200n8"
+    "androidboot.hardware=minnow"
+  ];
+}
diff --git a/devices/motorola-smelt/default.nix b/devices/motorola-smelt/default.nix
new file mode 100644
index 00000000..8dbdd63b
--- /dev/null
+++ b/devices/motorola-smelt/default.nix
@@ -0,0 +1,57 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "motorola-smelt";
+  mobile.device.identity = {
+    name = "Moto 360 (2nd Gen)";
+    manufacturer = "Motorola";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon 400 (APQ8026)
+    soc = "qualcomm-apq8026";
+    ram = 512;
+    screen = {
+      width = 360;
+      height = 325;
+      chinHeight = 35;
+      shape = "chin";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "smelt";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=smelt"
+    "msm_rtb.filter=0x237"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/oppo-beluga/default.nix b/devices/oppo-beluga/default.nix
new file mode 100644
index 00000000..c2cef490
--- /dev/null
+++ b/devices/oppo-beluga/default.nix
@@ -0,0 +1,57 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "oppo-beluga";
+  mobile.device.identity = {
+    name = "OPPO Watch (46mm)";
+    manufacturer = "OPPO";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon Wear 3100
+    soc = "qualcomm-sda429w";
+    ram = 1024 * 1;
+    screen = {
+      width = 402;
+      height = 476;
+      shape = "squircle";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "beluga";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "4096";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=beluga"
+    "msm_rtb.filter=0x237"
+    "ehci-hcd.park=3"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/polar-pike/default.nix b/devices/polar-pike/default.nix
new file mode 100644
index 00000000..94a2f0f8
--- /dev/null
+++ b/devices/polar-pike/default.nix
@@ -0,0 +1,54 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "polar-pike";
+  mobile.device.identity = {
+    name = "M600";
+    manufacturer = "Polar";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # MediaTek MT2601
+    soc = "mediatek-mt2601";
+    ram = 512;
+    screen = {
+      width = 240;
+      height = 240;
+      shape = "square";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = false;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "pike";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x04000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=tty0"
+    "androidboot.hardware=pike"
+  ];
+}
diff --git a/devices/samsung-rinato/default.nix b/devices/samsung-rinato/default.nix
new file mode 100644
index 00000000..30a23a62
--- /dev/null
+++ b/devices/samsung-rinato/default.nix
@@ -0,0 +1,54 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "samsung-rinato";
+  mobile.device.identity = {
+    name = "Gear 2";
+    manufacturer = "Samsung";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Samsung Exynos 3250
+    soc = "samsung-exynos3250";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 320;
+      shape = "square";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "rinato";
+    bootimg.flash = {
+      offset_base = "0x40000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttySAC2,115200"
+    "androidboot.hardware=rinato"
+  ];
+}
diff --git a/devices/samsung-sprat/default.nix b/devices/samsung-sprat/default.nix
new file mode 100644
index 00000000..55ce34d8
--- /dev/null
+++ b/devices/samsung-sprat/default.nix
@@ -0,0 +1,57 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "samsung-sprat";
+  mobile.device.identity = {
+    name = "Gear Live";
+    manufacturer = "Samsung";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon 400 (APQ8026)
+    soc = "qualcomm-apq8026";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 320;
+      shape = "square";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "sprat";
+    bootimg.flash = {
+      offset_base = "0x00000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=sprat"
+    "msm_rtb.filter=0x237"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/sony-tetra/default.nix b/devices/sony-tetra/default.nix
new file mode 100644
index 00000000..2984ba6c
--- /dev/null
+++ b/devices/sony-tetra/default.nix
@@ -0,0 +1,54 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "sony-tetra";
+  mobile.device.identity = {
+    name = "SmartWatch 3 (SWR50)";
+    manufacturer = "Sony";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Broadcom BCM23550
+    soc = "broadcom-bcm23550";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 320;
+      shape = "square";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = false;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "tetra";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyS0,115200"
+    "androidboot.hardware=tetra"
+  ];
+}
diff --git a/devices/ticwatch-catfish/default.nix b/devices/ticwatch-catfish/default.nix
new file mode 100644
index 00000000..7fc7fb2a
--- /dev/null
+++ b/devices/ticwatch-catfish/default.nix
@@ -0,0 +1,56 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "ticwatch-catfish";
+  mobile.device.identity = {
+    name = "TicWatch Pro";
+    manufacturer = "Mobvoi";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon Wear 2100 (MSM8909W)
+    soc = "qualcomm-msm8909w";
+    ram = 1024 * 1;
+    screen = {
+      width = 400;
+      height = 400;
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "catfish";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=catfish"
+    "msm_rtb.filter=0x237"
+    "ehci-hcd.park=3"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/ticwatch-harmony/default.nix b/devices/ticwatch-harmony/default.nix
new file mode 100644
index 00000000..9bff8de0
--- /dev/null
+++ b/devices/ticwatch-harmony/default.nix
@@ -0,0 +1,54 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "ticwatch-harmony";
+  mobile.device.identity = {
+    name = "KW68 / MTK6580 (Round)";
+    manufacturer = "Kingwear";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # MediaTek MT6580
+    soc = "mediatek-mt6580";
+    ram = 512;
+    screen = {
+      width = 400;
+      height = 400;
+      shape = "round";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = false;
+      burnInProtection = false;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "harmony";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x04000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=tty0"
+    "androidboot.hardware=harmony"
+  ];
+}
diff --git a/devices/ticwatch-inharmony/default.nix b/devices/ticwatch-inharmony/default.nix
new file mode 100644
index 00000000..7a9baf50
--- /dev/null
+++ b/devices/ticwatch-inharmony/default.nix
@@ -0,0 +1,54 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "ticwatch-inharmony";
+  mobile.device.identity = {
+    name = "MTK6580 (Square)";
+    manufacturer = "Generic";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # MediaTek MT6580
+    soc = "mediatek-mt6580";
+    ram = 512;
+    screen = {
+      width = 320;
+      height = 320;
+      shape = "square";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = false;
+      burnInProtection = false;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "inharmony";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x04000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=tty0"
+    "androidboot.hardware=inharmony"
+  ];
+}
diff --git a/devices/ticwatch-mooneye/default.nix b/devices/ticwatch-mooneye/default.nix
new file mode 100644
index 00000000..6a76cef2
--- /dev/null
+++ b/devices/ticwatch-mooneye/default.nix
@@ -0,0 +1,54 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "ticwatch-mooneye";
+  mobile.device.identity = {
+    name = "TicWatch E & S";
+    manufacturer = "Mobvoi";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # MediaTek MT2601
+    soc = "mediatek-mt2601";
+    ram = 512;
+    screen = {
+      width = 400;
+      height = 400;
+      shape = "round";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "mooneye";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x04000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=tty0"
+    "androidboot.hardware=mooneye"
+  ];
+}
diff --git a/devices/ticwatch-rubyfish/default.nix b/devices/ticwatch-rubyfish/default.nix
new file mode 100644
index 00000000..71420dce
--- /dev/null
+++ b/devices/ticwatch-rubyfish/default.nix
@@ -0,0 +1,56 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "ticwatch-rubyfish";
+  mobile.device.identity = {
+    name = "TicWatch Pro 3";
+    manufacturer = "Mobvoi";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon Wear 4100 (SDA429W / SW4100)
+    soc = "qualcomm-sda429w";
+    ram = 1024 * 1;
+    screen = {
+      width = 454;
+      height = 454;
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "rubyfish";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x01000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x00000100";
+      pagesize = "4096";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=rubyfish"
+    "msm_rtb.filter=0x237"
+    "ehci-hcd.park=3"
+    "vmalloc=300M"
+  ];
+}
diff --git a/devices/ticwatch-skipjack/default.nix b/devices/ticwatch-skipjack/default.nix
new file mode 100644
index 00000000..454a6f99
--- /dev/null
+++ b/devices/ticwatch-skipjack/default.nix
@@ -0,0 +1,56 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "ticwatch-skipjack";
+  mobile.device.identity = {
+    name = "TicWatch C2 / C2+";
+    manufacturer = "Mobvoi";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon Wear 2100 (MSM8909W)
+    soc = "qualcomm-msm8909w";
+    ram = 512;
+    screen = {
+      width = 360;
+      height = 360;
+      shape = "round";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "skipjack";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=skipjack"
+    "msm_rtb.filter=0x237"
+  ];
+}
diff --git a/devices/ticwatch-tunny/default.nix b/devices/ticwatch-tunny/default.nix
new file mode 100644
index 00000000..69a52c41
--- /dev/null
+++ b/devices/ticwatch-tunny/default.nix
@@ -0,0 +1,56 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  imports = [
+    ../families/asteroidos.nix
+  ];
+
+  mobile.device.name = "ticwatch-tunny";
+  mobile.device.identity = {
+    name = "TicWatch E2 / S2";
+    manufacturer = "Mobvoi";
+  };
+
+  mobile.device.supportLevel = "supported";
+
+  mobile.hardware = {
+    # Qualcomm Snapdragon Wear 2100 (MSM8909W)
+    soc = "qualcomm-msm8909w";
+    ram = 512;
+    screen = {
+      width = 400;
+      height = 400;
+      shape = "round";
+    };
+    watch = {
+      hasCrown = false;
+      hasHaptics = true;
+      aodSupport = true;
+      burnInProtection = true;
+    };
+  };
+
+  mobile.system.android = {
+    device_name = "tunny";
+    bootimg.flash = {
+      offset_base = "0x80000000";
+      offset_kernel = "0x00008000";
+      offset_ramdisk = "0x02000000";
+      offset_second = "0x00f00000";
+      offset_tags = "0x01e00000";
+      pagesize = "2048";
+    };
+  };
+
+  boot.kernelParams = [
+    "console=ttyMSM0,115200,n8"
+    "androidboot.console=ttyMSM0"
+    "androidboot.hardware=tunny"
+    "msm_rtb.filter=0x237"
+  ];
+}
